src/status_im2/contexts/chat/messages/link_preview/events_test.cljs

```diff
@@ -0,0 +1,139 @@
+(ns status-im2.contexts.chat.messages.link-preview.events-test
+  (:require [cljs.test :as t :include-macros true]
+            matcher-combinators.test
+            [status-im.mailserver.core :as mailserver]
+            [status-im2.contexts.chat.messages.link-preview.events :as sut]))
+
+(t/deftest fetch-community
+  (t/testing "with community id"
+    (t/testing "update resolving indicator in db"
+      (t/is (match?
+             {:db {:communities/resolve-community-info {"community-id" true}}}
+             (sut/fetch-community {} ["community-id"]))))
+    (t/testing "call the fetch community rpc method with correct community id"
+      (t/is (match?
+             {:json-rpc/call [{:method "wakuext_fetchCommunity"
+                               :params [{:CommunityKey    "community-id"
+                                         :TryDatabase     true
+                                         :WaitForResponse true}]}]}
+             (sut/fetch-community {} ["community-id"])))))
+  (t/testing "with nil community id"
+    (t/testing "do nothing"
+      (t/is (match?
+             nil
+             (sut/fetch-community {} nil))))))
+
+(t/deftest community-failed-to-resolve
+  (t/testing "given a community id"
+    (t/testing "remove community id from resolving indicator in db"
+      (t/is (match?
+             nil
+             (get-in (sut/community-failed-to-resolve {:db {:communities/resolve-community-info
+                                                            {"community-id" true}}}
+                                                      ["community-id"])
+                     [:db :communities/resolve-community-info "community-id"]))))))
+
+(t/deftest community-resolved
+  (with-redefs [sut/community-link (fn [id] (str "community-link+" id))]
+    (t/testing "given a community"
+      (let [cofx {:db {:communities/resolve-community-info {"community-id" true}}}
+            arg  ["community-id" {:id "community-id"}]]
+        (t/testing "remove community id from resolving indicator in db"
+          (t/is (match?
+                 nil
+                 (get-in (sut/community-resolved cofx arg)
+                         [:db :communities/resolve-community-info "community-id"]))))
+        (t/testing "dispatch fxs"
+          (t/is (match?
+                 {:fx [[:dispatch [:communities/handle-community {:id "community-id"}]]
+                       [:dispatch
+                        [:chat.ui/cache-link-preview-data "community-link+community-id"
+                         {:id "community-id"}]]]}
+                 (sut/community-resolved cofx arg))))))
+    (t/testing "given a joined community"
+      (let [cofx {:db {:communities/resolve-community-info {"community-id" true}}}
+            arg  ["community-id" {:id "community-id" :joined true}]]
+        (t/testing "dispatch fxs, do not spectate community"
+          (t/is (match?
```

Comment:
Real nice! Was `match?` useful in your testing endeavor @yqrashawn ?

Reply:
super useful